File: gof-design-patterns/srcs/P00_01_Encapsulation/BankAccount.java
package P00_01_Encapsulation;

public class BankAccount {
    private double balance;  // データをprivateで隠蔽
    
    public BankAccount(double initialBalance) {
        this.balance = initialBalance;
    }
    
    public void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
        }
    }
    
    public void withdraw(double amount) {
        if (amount > 0 && amount <= balance) {
            balance -= amount;
        }
    }
    
    public double getBalance() {
        return balance;
    }
    
    public static void main(String[] args) {
        BankAccount account = new BankAccount(1000);
        account.deposit(500);
        account.withdraw(200);
        System.out.println("残高: " + account.getBalance());
    }
}

File: gof-design-patterns/typescripts/P00_01_Encapsulation/BankAccount.js
var BankAccount = /** @class */ (function () {
    function BankAccount(initialBalance) {
        this.balance = initialBalance;
    }
    BankAccount.prototype.deposit = function (amount) {
        if (amount > 0) {
            this.balance += amount;
        }
    };
    BankAccount.prototype.withdraw = function (amount) {
        if (amount > 0 && amount <= this.balance) {
            this.balance -= amount;
        }
    };
    BankAccount.prototype.getBalance = function () {
        return this.balance;
    };
    return BankAccount;
}());
// 実行例
var account = new BankAccount(1000);
account.deposit(500);
account.withdraw(200);
console.log("\u6B8B\u9AD8: ".concat(account.getBalance()));
